src/main/java/uk/gov/pay/directdebit/common/validation/ApiValidation.java

```diff
@@ -19,18 +19,54 @@
     private static final Logger LOGGER = PayLoggerFactory.getLogger(ApiValidation.class);
 
     private String[] requiredFields;
+    private Map<String, Integer> minimumFieldsSize;
     private Map<String, Integer> maximumFieldsSize;
     private Map<String, Function<String, Boolean>> validators;
 
-    public ApiValidation(String[] requiredFields, Map<String, Integer> maximumFieldsSize, Map<String, Function<String, Boolean>> validators) {
+    public ApiValidation(String[] requiredFields,
+                         Map<String, Integer> minimumFieldsSize,
+                         Map<String, Integer> maximumFieldsSize,
+                         Map<String, Function<String, Boolean>> validators) {
         this.requiredFields = requiredFields;
+        this.minimumFieldsSize = minimumFieldsSize;
         this.maximumFieldsSize = maximumFieldsSize;
         this.validators = validators;
     }
 
-    private boolean isFieldSizeValid(Map<String, String> chargeRequest, String fieldName, int fieldSize) {
-        Optional<String> value = Optional.ofNullable(chargeRequest.get(fieldName));
-        return !value.isPresent() || value.get().length() <= fieldSize; //already checked that mandatory fields are already there
+    public void validate(String externalId, Map<String, String> request) {
+        List<String> missingFields = checkMissingFields(request, requiredFields);
+        if (!missingFields.isEmpty()) {
+            LOGGER.error("Error validating request {}, missing mandatory fields", externalId);
+            throw new MissingMandatoryFieldsException(missingFields);
+        }
+        List<String> invalidSizeFields;
+        invalidSizeFields = checkInvalidMaximumSizeFields(request, maximumFieldsSize);
+        if (!invalidSizeFields.isEmpty()) {
+            LOGGER.error("Error validating request {}, the size of a field(s) is too long", externalId);
+            throw new InvalidSizeFieldsException(invalidSizeFields);
+        }
+        invalidSizeFields = checkInvalidMinimumSizeFields(request, minimumFieldsSize);
+        if (!invalidSizeFields.isEmpty()) {
+            LOGGER.error("Error validating request {}, the size of a field(s) is too long", externalId);
+            throw new InvalidSizeFieldsException(invalidSizeFields);
+        }
+        List<String> invalidFields = validateParams(request, validators);
+        if (!invalidFields.isEmpty()) {
+            LOGGER.error("Error validating request {}, fields are invalid", externalId);
+            throw new InvalidFieldsException(invalidFields);
+        }
+    }
+
+    public void validate(Map<String, String> request) {
+        validate("<external id not yet created>", request);
+    }
+
+    public static boolean isNumeric(String string) {
+        return (string != null) && string.matches("[0-9]+");
+    }
+
+    public static boolean isNotNullOrEmpty(String string) {
+        return !((string == null) || string.isEmpty());
```

Comment:
We can use StringUtils.isBlank(`<string>`) (or the equivalent guava helpers) to make it a bit easier to read